### PR TITLE
Resource manager can return an empty list

### PR DESCRIPTION
**Pull request by ohadbitt**
Also added test for so it does not return old storage

### CODE REVIEW

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java [hunk 1]
```diff
@@ -61,13 +60,9 @@ public class ResourceManager implements Closeable, IngestionResourceManager {
...
     private String identityToken;
-    private IngestionResource<ContainerWithSas> containers;
-    private IngestionResource<TableWithSas> statusTable;
-    private IngestionResource<QueueWithSas> queues;
-    private IngestionResource<QueueWithSas> successfulIngestionsQueues;
-    private IngestionResource<QueueWithSas> failedIngestionsQueues;
+    private IngestionResourceSet ingestionResourceSet;
 
-    ResourceManager(Client client, long defaultRefreshTime, long refreshTimeOnFailure, @Nullable CloseableHttpClient httpClient) {
+    public ResourceManager(Client client, long defaultRefreshTime, long refreshTimeOnFailure, @Nullable CloseableHttpClient httpClient) {
```
Comment (AsafMah) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java:65]: Why are you making it public? this means we are responsible to keep it not broken

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java [hunk 2]
```diff
@@ -409,4 +403,12 @@ boolean empty() {
             return this.resourcesList.size() == 0;
         }
     }
+
+    private static class IngestionResourceSet {
```
Comment (AsafMah) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java:407]: To be clear - the problem we had before is that we replaced the queues one at a time, and this fixes at by replacing them all at once?

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java (hunk 1 above)
Comment (ohadbitt) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java:65]: for some tests
will revert

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java (hunk 2 above)
Comment (ohadbitt) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java:407]: Not exactly 
we were setting this.containers for example to a new empty list 
while it was not lock protected and could be read empty at this point and even not reverted back on failure too ...

**Review by AsafMah — APPROVED** [review]

**Pull request merged**